From 5bd32357c12bc3fc5d8ae3dfafc5f7207c49cda2 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Fri, 14 Jan 2022 13:55:32 +0100
Subject: [PATCH 60/69] Revert "bfq: fix use-after-free in
 bfq_dispatch_request"

This reverts commit a8679b6a74be63fe0030f8add7ef50e07204ec6f.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-iosched.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index 9ec9d5dda..4396abf92 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -5227,7 +5227,7 @@ static struct request *bfq_dispatch_request(struct blk_mq_hw_ctx *hctx)
 	struct bfq_data *bfqd = hctx->queue->elevator->elevator_data;
 	struct request *rq;
 	struct bfq_queue *in_serv_queue;
-	bool waiting_rq, idle_timer_disabled = false;
+	bool waiting_rq, idle_timer_disabled;
 
 	spin_lock_irq(&bfqd->lock);
 
@@ -5237,10 +5237,9 @@ static struct request *bfq_dispatch_request(struct blk_mq_hw_ctx *hctx)
 
 	rq = __bfq_dispatch_request(hctx);
 
-	if (in_serv_queue == bfqd->in_service_queue) {
-		idle_timer_disabled =
-			waiting_rq && !bfq_bfqq_wait_request(in_serv_queue);
-	}
+	idle_timer_disabled =
+		waiting_rq && !bfq_bfqq_wait_request(in_serv_queue);
+
 	spin_unlock_irq(&bfqd->lock);
 
 	bfq_update_dispatch_stats(hctx->queue, rq, in_serv_queue,
-- 
2.34.1.75.gabe6bb3905

